tests/zsa.rs

```diff
@@ -439,10 +450,53 @@ fn zsa_issue_and_transfer() {
                     asset: zsa_spend_t7_2.note.asset(),
                 },
             ],
+            vec![],
             anchor_t7,
             4,
             &keys,
         );
     });
     assert!(result.is_err());
+
+    // 9. Burn ZSA assets
+    build_and_verify_bundle(
+        vec![&zsa_spend_1],
+        vec![],
+        vec![(
+            zsa_spend_1.note.asset(),
+            zsa_spend_1.note.value().inner() as i64,
+        )],
+        anchor,
+        2,
+        &keys,
+    );
+
+    // 10. Burn a part of ZSA assets
+    let value_to_burn: i64 = 3;
+    let value_to_transfer: u64 = zsa_spend_1.note.value().inner() - value_to_burn as u64;
+
+    build_and_verify_bundle(
+        vec![&zsa_spend_1],
+        vec![TestOutputInfo {
+            value: NoteValue::from_raw(value_to_transfer),
+            asset: zsa_spend_1.note.asset(),
+        }],
+        vec![(zsa_spend_1.note.asset(), value_to_burn)],
+        anchor,
+        2,
+        &keys,
+    );
+
+    // 11. Try to burn native asset - should fail
+    let result = std::panic::catch_unwind(|| {
```

Comment:
Not sure that we need `catch_unwind` here. Just test result for error.

Reply:
That is, `build_and_verify_bundle()` should populate the `Result<()>` (and `Err` implicitly) to this layer.
Otherwise, this is not a very precise test.